ckanext/knowledgehub/logic/action/delete.py

```diff
@@ -497,13 +498,48 @@ def package_delete(context, data_dict):
     :type id: string
 
     '''
-
+    id = data_dict.get('id')
+    try:
+        result = toolkit.get_action('search_visualizations')(context, {
+        'text': '*',
+        'fq': [
+            '+khe_package_id:{0}'.format(data_dict['id']),
+        ]})
+        if result['count'] > 0:
+                for res in result['results']:
+                    resource_view_delete(context, { 'id': res['id'] })
+    except Exception as e:
+        print str(e)
     ckan_package_delete(context, data_dict)
     try:
         KWHData.delete({'dataset': data_dict['id']})
     except Exception as e:
         log.debug('Cannot remove dataset from kwh data %s' % str(e))
 
+def resource_delete(context, data_dict):
+    '''Delete a resource from a dataset.
+
+    You must be a sysadmin or the owner of the resource to delete it.
+
+    :param id: the id of the resource
+    :type id: string
+
+    '''
+
+    id = data_dict.get('id')
+    try:
+        result = toolkit.get_action('search_visualizations')(context, {
+        'text': '*',
+        'fq': [
+            '+khe_resource_id:{0}'.format(data_dict['id']),
+        ]})
+        if result['count'] > 0:
+            for res in result['results']:
+                resource_view_delete(context, { 'id': res['id'] })
+    except Exception as e:
+        print str(e)
```

Comment:
Just log this one using `log.error()` and  `log.exception()` so it logs full stack trace